projects/step-core/src/lib/directives/test-dataid.directive.ts

```diff
@@ -0,0 +1,14 @@
+import { Directive, ElementRef, inject, Input } from '@angular/core';
+
+@Directive({
+  selector: '[testDataId]',
+})
+export class TestDataIdDirective {
+  @Input('testDataId') testDataId: string | undefined;
+
+  private _el = inject(ElementRef);
+
+  ngOnInit() {
+    this._el.nativeElement.setAttribute('data-testid', this.testDataId);
```

Comment:
1. `data-testid` name looks very common. I would also renamed it to `data-step-testid`, to specify explicitly that it is a **step** testing identifier.
2. It's recommended to use [Renderer](https://angular.io/api/core/Renderer2#description) service for manipulation with DOM

Reply:
@lukasz-lepa @dvladir I discussed this and thought about it and finally agree, lets rename to data-step-testid